### PR TITLE
add test Testing Guide

### PR DESCRIPTION
**Pull request by harryEth**
### 📦 Pull Request

Add testing guide

### 🗜 Versioning

(Check _one!_)

- [x] Patch: Bug Fix?
- [ ] Minor: New Feature?
- [ ] Major: Breaking Change?

### ⚠️ Update `CHANGELOG.md`

- [ ] I have updated the `Upcoming Changes` section of `CHANGELOG.md` with context related to this Pull Request.

### CODE REVIEW

**Comment by harryEth** [comment]
[ch25097]

File: README.md [hunk 1]
```diff
@@ -42,6 +42,12 @@ Alternatively, you can load via CDN with by adding a script tag to your app’s
 <script src="https://cdn.jsdelivr.net/npm/magic-sdk/dist/magic.js"></script>
 ```
 
+## 🔗 Testing
+Run test
+```bash
+yarn test
+```
+
```
Comment (smithki) [inline README.md:50]: Let's add a bit more detail here:

Testing can also be on a per-package basis, like this:

```bash
PKG=magic-sdk yarn test
PKG=@magic-sdk/react-native yarn test
```

You can also glob the files you want to test specifically:

```bash
yarn test /test/**/constructor.spec.ts
```

**Review by smithki — DISMISSED** [review]
Just nits, then 🐑

File: README.md [hunk 2]
```diff
@@ -43,11 +43,22 @@ Alternatively, you can load via CDN with by adding a script tag to your app’s
 ```
 
 ## 🔗 Testing
-Run test
+Run test for all packages
```
Comment (smithki) [inline README.md:46]: "Run test**s** for all packages"

File: README.md [hunk 3]
```diff
@@ -43,11 +43,22 @@ Alternatively, you can load via CDN with by adding a script tag to your app’s
...
 ```bash
 yarn test
 ```
 
+Test individual package
```
Comment (smithki) [inline README.md:51]: "Test **an** individual package"

File: README.md [hunk 4]
```diff
@@ -43,11 +43,22 @@ Alternatively, you can load via CDN with by adding a script tag to your app’s
...
+```bash
+PKG=magic-sdk yarn test
+PKG=@magic-sdk/react-native yarn test
+```
+
+Test specific file
```
Comment (smithki) [inline README.md:57]: "Test specific file**s**"

**Review by smithki — DISMISSED** [review]
Just a few last nits then we are good :)

**Review by smithki — APPROVED** [review]

**Comment by smithki** [comment]
:rocket: PR was released in [`@magic-sdk/provider@3.0.1`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/provider%403.0.1), [`@magic-sdk/react-native@3.0.1`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/react-native%403.0.1), [`@magic-sdk/types@2.0.1`](https://github.com/magiclabs/magic-js/releases/tag/%40magic-sdk/types%402.0.1), [`magic-sdk@3.0.1`](https://github.com/magiclabs/magic-js/releases/tag/magic-sdk%403.0.1) :rocket:

**Pull request merged**